### PR TITLE
Cppref Rewrite

### PR DESCRIPTION
**Pull request by SqueakyBeaver**
This fixes the `std::cpp::string`, `c/string/byte/atof`, and similar redundancy errors as well as displaying such results as `std::string` twice. This will also fix some things such as me forgetting to change `print` to `await ctx.send`.

Edit:
TODO
- [x] Fix things such as `std::io::cout`

- [x] Use a different method for searching

### CODE REVIEW

File: src/cogs/cpp.py [hunk 1]
```diff
@@ -52,28 +52,22 @@ async def cppref(self, ctx, *, query: str):
...
-            # print(check_name)
 
-            # print(check_name)
             f_name = check_name.replace("/", "::")
             f_name = f_name.replace(".html", "")
 
             if check_name.startswith(("language", "concept")) and not check_name.startswith("concepts"):
                 special_pages.append(
-                    f'[`{f_name}`]({result})')
+                    f'[`{f_name.replace("cpp::", "")}`]({result})')
```
Comment (Yuhanun) [inline src/cogs/cpp.py]: it's not just cpp::, there's a lot more. 
image

Comment (Yuhanun) [reply src/cogs/cpp.py]: So if it starts with std::{directory} -> replace

**Review by Yuhanun — CHANGES_REQUESTED** [review]

File: src/cogs/cpp.py (hunk 1 above)
Comment (Yuhanun) [reply src/cogs/cpp.py]: What we could do is scan the directory for other directories, and add it to a list

Comment (SqueakyBeaver) [reply src/cogs/cpp.py]: I hadn't thought of that. Upon further investigation, I have found that there is a `header` folder that contains the names of all the headers. So if the name is a file in that folder, I can remove it from the list. But then I run into the problem of stuff like `std::string`. There is a header `string` and a class `string`. I will have to think about a solution to this. If you have any ideas, feel free to share!

Comment (Yuhanun) [reply src/cogs/cpp.py]: what if it just starts with std::string, then you can just remove it no? like std::string::string would be the result no?

**Comment by SqueakyBeaver** [comment]
It should be ready now. The search results are now displayed such as `(header) string` and `(string) std::basic_string` for C++ searches and `(header) string` and `(string) byte` (the "normal" `string` functions in C are on the [byte string](https://en.cppreference.com/w/c/string/byte) page). I think this clears up any confusion of where the presented items are located.
This can be seen in the screenshots:
cref
cppref

**Comment by Yuhanun** [comment]
Actually, experimental is part of the library, not the language 🥺 I'm sorry but could you change that?

**Comment by SqueakyBeaver** [comment]
Yeah, I kind of nuked experimental from the search because of the `std::experimental::...`, but I think I could add a list of the headers that have repeated namespaces (all I can think of are `experimental` and `chrono`) and just say that the results are not 100% accurate because of the way cppreference named their links (`experimental/fs` instead of `experimental/filesystem` is my biggest worry)

**Comment by SqueakyBeaver** [comment]
nested namespaces
^ Example of the nested namespaces. I will continue to add more if needed

**Review by Yuhanun — APPROVED** [review]

**Pull request merged**